components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/cmd/FunctionCmd.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.apimgt.gateway.cli.cmd;
+
+import com.beust.jcommander.JCommander;
+import com.beust.jcommander.Parameter;
+import com.beust.jcommander.Parameters;
+import org.apache.commons.io.FileUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.wso2.apimgt.gateway.cli.exception.CLIRuntimeException;
+import org.wso2.apimgt.gateway.cli.utils.GatewayCmdUtils;
+import org.wso2.apimgt.gateway.cli.utils.RouteUtils;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.util.List;
+
+@Parameters(commandNames = "function", commandDescription = "add function")
+public class FunctionCmd implements GatewayLauncherCmd {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(FunctionCmd.class);
+    private static PrintStream outStream = System.out;
+
+    @Parameter(hidden = true, required = true)
+    private List<String> mainArgs;
+
+    @Parameter(names = {"-a", "--add"}, hidden = true)
```

Comment:
Use `-a` to refer API object. Since that is common between all commands. We can use `-A` for add and `-U` for update to keep the consistancy.
```suggestion
    @Parameter(names = {"-A", "--add"}, hidden = true)
```

Reply:
changes are added